File: src/main/java/com/stainberg/keditview/KoalaImageLoadPoll.java
package com.stainberg.keditview;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * Created by Stainberg on 7/6/17.
 */

class KoalaImageLoadPoll {

    private static KoalaImageLoadPoll instance;
    private ExecutorService poll;

    private KoalaImageLoadPoll() {
        poll = Executors.newCachedThreadPool();
    }

    public static KoalaImageLoadPoll getPoll() {
        if(instance == null) {
            synchronized (KoalaImageLoadPoll.class) {
                if(instance == null) {
                    instance = new KoalaImageLoadPoll();
                }
            }
        }
        return instance;
    }

    public void handle(Runnable r) {
        poll.execute(r);
    }

}

File: src/main/java/com/stainberg/keditview/KoalaImageLoadPoll.kt
package com.stainberg.keditview

import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

/**
 * Created by Stainberg on 7/6/17.
 */

internal class KoalaImageLoadPoll private constructor() {
    private val poll : ExecutorService

    init {
        poll = Executors.newCachedThreadPool()
    }

    fun handle(r : Runnable) {
        poll.execute(r)
    }

    companion object {

        private var instance : KoalaImageLoadPoll? = null

        fun getPoll() : KoalaImageLoadPoll {
            if (instance == null) {
                synchronized(KoalaImageLoadPoll::class.java) {
                    if (instance == null) {
                        instance = KoalaImageLoadPoll()
                    }
                }
            }
            return instance!!
        }
    }

}
